python-package/lightgbm/basic.py

```diff
@@ -837,52 +838,98 @@ class _InnerPredictor:
-            Path to the model file.
-        booster_handle : object or None, optional (default=None)
+        booster : Booster
             Handle of Booster.
-        pred_parameter: dict or None, optional (default=None)
+        pandas_categorical : list of list, or None
+            If provided, list of categories for ``pandas`` categorical columns.
+            Where the ``i``th element of the list contains the categories for the ``i``th categorical feature.
+        pred_parameter: dict
             Other parameters for the prediction.
+        manage_handle : bool
+            If ``True``, free the corresponding Booster on the C++ side when this Python object is deleted.
         """
-        self._handle = ctypes.c_void_p()
-        self.__is_manage_handle = True
-        if model_file is not None:
-            """Prediction task"""
-            out_num_iterations = ctypes.c_int(0)
-            _safe_call(_LIB.LGBM_BoosterCreateFromModelfile(
-                _c_str(str(model_file)),
-                ctypes.byref(out_num_iterations),
-                ctypes.byref(self._handle)))
-            out_num_class = ctypes.c_int(0)
-            _safe_call(_LIB.LGBM_BoosterGetNumClasses(
-                self._handle,
-                ctypes.byref(out_num_class)))
-            self.num_class = out_num_class.value
-            self.num_total_iteration = out_num_iterations.value
-            self.pandas_categorical = _load_pandas_categorical(file_name=model_file)
-        elif booster_handle is not None:
-            self.__is_manage_handle = False
-            self._handle = booster_handle
-            out_num_class = ctypes.c_int(0)
-            _safe_call(_LIB.LGBM_BoosterGetNumClasses(
+        self._handle = booster_handle
+        self.__is_manage_handle = manage_handle
+        self.pandas_categorical = pandas_categorical
+        self.pred_parameter = _param_dict_to_str(pred_parameter)
+
+        out_num_class = ctypes.c_int(0)
+        _safe_call(
+            _LIB.LGBM_BoosterGetNumClasses(
                 self._handle,
-                ctypes.byref(out_num_class)))
-            self.num_class = out_num_class.value
-            self.num_total_iteration = self.current_iteration()
-            self.pandas_categorical = None
-        else:
-            raise TypeError('Need model_file or booster_handle to create a predictor')
+                ctypes.byref(out_num_class)
+            )
+        )
+        self.num_class = out_num_class.value
 
-        pred_parameter = {} if pred_parameter is None else pred_parameter
-        self.pred_parameter = _param_dict_to_str(pred_parameter)
+    @classmethod
+    def from_booster(
+        cls,
+        booster: "Booster",
+        pred_parameter: Dict[str, Any]
+    ) -> "_InnerPredictor":
+        """Initialize an ``_InnerPredictor`` from a ``Booster``.
+
+        Parameters
+        ----------
+        booster : Booster
+            Handle of Booster.
```

Comment:
I know this looks very repetitive but this isn't just the handle, right? It's a Booster object
```suggestion
            Booster.
```

Reply:
Yep you're totally right. In the middle of doing this, I decided it'd be simpler for `.from_booster()` to take a `Booster` object instead of a handle... but then clearly missed updates to some of the docs.

Sorry for missing that, and thanks for catching them!